trace/sampling.go

```diff
@@ -24,12 +24,6 @@ func newDefaultSampler() Sampler {
 	return ProbabilitySampler(defaultSamplingProbability)
 }
 
-// SetDefaultSampler sets the default sampler used when creating new spans.
-//
-// Deprecated: Use ApplyConfig.
-func SetDefaultSampler(sampler Sampler) {
```

Comment:
This might be too early to remove, there are a bunch of examples and running code in the wild using SetDefaultSampler, perhaps let's leave this one in for a bit, or did we already remove it? I think changing it recently was a mistake as we should give some sunsetting time.

Reply:
Alright then, but before we merge this PR, we'll need to perhaps fix Google Cloud
https://github.com/GoogleCloudPlatform/google-cloud-go/blob/01dbcbbaf95d5ec2b07fe6dcd7db442be0f5b0c4/internal/testutil/go18.go#L42
and perhaps ping a few other folks such as https://blog.doit-intl.com/measure-once-export-anywhere-opencensus-in-the-wild-61724f44eb00 and also my examples.

Reply:
I'll mail the one for cloud-go.

Reply:
I've mailed https://code-review.googlesource.com/#/c/gocloud/+/26130 but there is trouble logging in :)